### ISSUE TITLE
Using Arduino Nano to communicate with VESC

### ISSUE THREAD

**Issue by SolidGeek** [bastianraschke/ESP8266VESC#1]
Hi, great library! 

I am having trouble understanding if this can be used with an ordinary Arduino Nano or not. Using software serial to communicate doesn't seem to work, with an Arduino. 

I have tried to connect an Arduino nano and used your sketch ReadVESCValues, however, it keeps saying "The packet CRC value is different (corrupted packet)!". 

Am I right to believe this is caused by the lack of buffer with SoftwareSerial? Or does it not work with a VESC 6? 

Thanks!

**Comment by bastianraschke** [comment]
Hmm, according to some articles on the internet, the VESC 6 seems to be a totally new product including the software. I can't find any official stuff on Benjamins website regarding the VESC 6 for now.

Because the CRC check is the last step, i guess the software serial communication works fine, but maybe you can test PIN 10 and 11 for software serial as last chance. But i guess, Benjamin changed the protocol a bit that causes the trouble. But i just have my VESC 4 here, so i'm not able to test it.

**Comment by SolidGeek** [comment]
Thanks for the fast response, means a lot 👍 ! 

Unfortunately, I am not able to test with PIN 10 and 11 at the moment because pin 10 and 11 is used by my nRF24 receiver. I believe you are right, that the problem is most likely caused by the changes made in VESC 6. I know RollingGecko has a working branch for VESC 6, however, is it not as clean and easy to use as your library is.

**Comment by bastianraschke** [comment]
Yeah i'm pretty sure, the problems are caused by a protocol change. It would be quite hard for me to develop for a platform without a device to test my implementation ;-)
